eZ/Publish/Core/REST/Common/FieldTypeProcessor/RelationProcessor.php

```diff
@@ -37,7 +37,7 @@ public function postProcessFieldSettingsHash($outgoingSettingsHash)
         $outgoingSettingsHash = parent::postProcessFieldSettingsHash($outgoingSettingsHash);
 
         if (isset($outgoingSettingsHash['selectionRoot'])) {
-            $outgoingSettingsHash['selectionRootHref'] = $this->mapToContentHref(
+            $outgoingSettingsHash['selectionRootHref'] = $this->mapToLocationHref(
                 $outgoingSettingsHash['selectionRoot']
```

Comment:
selectionRoot only contains a location id, so you are right `selectionRootHref` won't have the expected value